src/types.ts

```diff
@@ -114,6 +115,7 @@ export type SchemaAttribute = {
   required?: boolean;
   matches?: SchemaMatches | ((config: Config) => SchemaMatches);
   errorLevel?: ValidationError['level'];
+  validate?(value: Value, config: Config): ValidationError[];
```

Comment:
Hmm. I feel like we should use use the `validate` function that exists within `type` instead of adding a new one here. Is there something this supports that custom attribute types doesn't?